File: algorithms/1_easy/N/01725_number_of_rectangles_that_can_form_the_largest_square/solution.py
class Solution:
    def countGoodRectangles(self, rectangles: List[List[int]]) -> int:
        count = 0
        max_len = -1

        for rec in rectangles:
            curr_len = min(rec[0], rec[1])

            if max_len < curr_len:
                max_len = curr_len
                count = 1
            elif max_len == curr_len:
                count += 1

        return count

File: algorithms/1_easy/N/01725_number_of_rectangles_that_can_form_the_largest_square/solution.cpp
class Solution {
public:
    int countGoodRectangles(vector<vector<int>>& rectangles) {
        int max_len{-1};
        int count{0};

        for (const auto &rec: rectangles) {
            int curr_len{min(rec[0], rec[1])};

            if (max_len < curr_len) {
                max_len = curr_len;
                count = 1;
            } else if (max_len == curr_len) {
                count += 1;
            }
        }

        return count;
    }
};
